### PR TITLE
Set firefox preference through the CLI #268

### PR DESCRIPTION
**Pull request by soulgalore**

### CODE REVIEW

File: lib/core/webdriver/firefox.js [hunk 1]
```diff
@@ -211,6 +211,14 @@ module.exports.configureBuilder = function(builder, options) {
     profile.setPreference(pref, defaultFirefoxPreferences[pref]);
   });
 
+  if (options.firefox && options.firefox.preference) {
```
Comment (tobli) [inline lib/core/webdriver/firefox.js]: Move this down, below ```const firefoxConfig = options.firefox || {};```

File: lib/core/webdriver/firefox.js [hunk 2]
```diff
@@ -211,6 +211,14 @@ module.exports.configureBuilder = function(builder, options) {
...
+    const userPrefs = Array.isArray(options.firefox.preference) ? options.firefox.preference : [options.firefox.preference];
```
Comment (tobli) [inline lib/core/webdriver/firefox.js]: Could you extract ```toArray```from engine.js to a util model, and use that here too? Then you can skip the if statement altogether, just
```for (const pref of toArray(firefoxConfig.preference))```

File: lib/core/webdriver/firefox.js [hunk 3]
```diff
@@ -211,6 +211,14 @@ module.exports.configureBuilder = function(builder, options) {
...
+    for (const pref of userPrefs) {
+        const nameAndValue = pref.split(':');
+        profile.setPreference(nameAndValue[0], nameAndValue[1]);
```
Comment (tobli) [inline lib/core/webdriver/firefox.js]: Should we have a check that nameAndValue has length 2?

File: lib/core/webdriver/firefox.js (hunk 1 above)
Comment (soulgalore) [inline lib/core/webdriver/firefox.js]: adding just before so we can override all values but still all prefreences on the same place.

File: lib/core/webdriver/firefox.js (hunk 3 above)
Comment (soulgalore) [inline lib/core/webdriver/firefox.js]: yep, let me fix that.

File: lib/core/webdriver/firefox.js (hunk 2 above)
Comment (soulgalore) [inline lib/core/webdriver/firefox.js]: I'll move that.

File: lib/core/webdriver/firefox.js [hunk 4]
```diff
@@ -242,6 +236,18 @@ module.exports.configureBuilder = function(builder, options) {
...
+        const nameAndValue = pref.split(':');
+        if (nameAndValue.length === 2) {
+          profile.setPreference(nameAndValue[0], nameAndValue[1]);
+        } else {
+          log.error('Firefox preferences %s need to of the format key:value, preference was not set', pref)
+        }
+    }
+  }
+
   const firefoxConfig = options.firefox || {};
```
Comment (tobli) [inline lib/core/webdriver/firefox.js]: If you move this just above the if statement, there's no need for ```if (options.firefox```. Also util.toArray handles undefined, to you can skip the if altogether and just have ```const userPrefs = util.toArray(firefoxConfig.preference);```

**Pull request merged**